content/en/docs/Writing policies/match-exclude.md

```diff
@@ -6,14 +6,26 @@ weight: 2
 
 The `match` and `exclude` filters control which resources policies are applied to.
 
-The `match` and `exclude` clauses have the same structure and can each contain the following elements:
+The `match` and `exclude` clauses have the same structure and can each contain **only one** of the two elements:
+
+* `any`: specify [resource filters](#resource-filters) on which Kyverno will perform the logical **OR** operation while choosing resources
+* `all`: specify [resource filters](#resource-filters) on which Kyverno will perform the logical **AND** operation while choosing resources
+
+## Resource Filters
+
+The following resource filters can be specified under an `any` or `all` clause.
 
 * `resources`: select resources by names, namespaces, kinds, label selectors, annotations, and namespace selectors.
 * `subjects`: select users, user groups, and service accounts
 * `roles`: select namespaced roles
 * `clusterRoles`: select cluster wide roles
 
-At least one element must be specified in a `match.resources.kinds` or `exclude` block. The `kind` attribute is mandatory when working with the `resources` element. Wildcards (`*`) are currently not supported in the `match.resources.kinds` field.
+
+{{% alert title="Note" color="info" %}}
+Specifying resource filters directly under match and exclude has been marked for deprecation and will be removed in a future release. It is highly recommended you specify them under `any` or `all` blocks.
+{{% /alert %}}
+
+At least one element must be specified in a `match.(any/all).resources.kinds` or `exclude` block. The `kind` attribute is mandatory when working with the `resources` element. Wildcards (`*`) are currently not supported in the `match.(any/all).resources.kinds` field.
```

Comment:
>The `kind` attribute is mandatory when working with the `resources` element.

@RinkiyaKeDad - is the above statement still valid? I thought we marked `match.resources` as optional and only require `kinds` if `match.resources` is used.

Quoting the comment from @windowsrefund:
>Also, the docs on https://kyverno.io/docs/writing-policies/match-exclude/#exclude-kube-system-namespace now say "The kind attribute is mandatory when working with the resources element". That's really confusing because one would think if that's true, each resources element under exclude.any would need to repeat the list of kinds in match.any.resources